pulpcore/app/models/publication.py

```diff
@@ -69,6 +72,25 @@ def create(cls, repository_version, pass_through=False):
             Adds a Task.created_resource for the publication.
         """
         with transaction.atomic():
+            # Check if at least some remote artifacts have allowed checksum
+            for ca in ContentArtifact.objects.filter(
+                content__in=repository_version.content, artifact=None
+            ):
+                allowed_checksum_content = False
+                remote_artifacts = ca.remoteartifact_set.all()
+                for checksum in Artifact.DIGEST_FIELDS:
+                    if any([ra for ra in remote_artifacts if getattr(ra, checksum, None)]):
+                        allowed_checksum_content = True
+                        break
+                if not allowed_checksum_content:
+                    raise ValueError(
```

Comment:
I found out that there's a specific exception for unrecognized / forbidden checksums:
```suggestion
                    raise UnsupportedDigestValidationError(
```